webapp/src/js/enketo/medic-xpath-extensions.js

```diff
@@ -14,11 +14,54 @@ var getValue = function(resultObject) {
   return resultObject.v;
 };
 
+var now_and_today = function() { return { t: 'date', v: new Date() }; };
+
+Date.prototype.toISOLocalString = function() {
+    //2012-09-05T12:57:00.000-04:00 (ODK)
+
+    if ( this.toString() === 'Invalid Date' ) {
+        return this.toString();
+    }
+
+    var dt = new Date( this.getTime() - ( this.getTimezoneOffset() * 60 * 1000 ) ).toISOString()
+        .replace( 'Z', this.getTimezoneOffsetAsTime() );
+
+    if ( dt.indexOf( 'T00:00:00.000' ) > 0 ) {
+        return dt.split( 'T' )[ 0 ];
+    } else {
+        return dt;
+    }
+};
+
+Date.prototype.getTimezoneOffsetAsTime = function() {
+    var offsetMinutesTotal;
+    var hours;
+    var minutes;
+    var direction;
```

Comment:
Don't declare variables without initialising them unless you have to. It's a waste of lines making it harder to read, but it also means you can't use `const`.